server/src/main/java/org/candlepin/controller/OwnerManager.java

```diff
@@ -288,15 +291,18 @@ public void refreshContentAccessMode(OwnerServiceAdapter adapter, Owner owner) {
     @Transactional
     public void refreshOwnerForContentAccess(Owner owner) {
         // we need to update the owner's consumers if the content access mode has changed
-        owner = ownerCurator.lockAndLoad(owner);
+        Owner locked = ownerCurator.lockAndLoad(owner);
+        if (locked == null) {
+            throw new IllegalStateException("Unable to obtain exclusive lock for owner: " + owner);
+        }
```

Comment:
Checking if the locked entity is null and throwing an ISE seems to be a common pattern. Does it make sense to have Curator.lockAndLoad(Entity e)  throw the exception instead of having to do the null check wherever it is called?

Reply:
No, because it's not a universal thing. In a couple of those cases, a null output could probably fall back to "do nothing" behavior safely, but given that it was written with the intent to always do something, I felt an exception would be a better drop-in replacement than silently skipping it. At least for now.

There are a couple cases (as you noticed in other comments) where we don't care if the object comes back or not.

Reply:
ACK